change settingpage

KJay221 opened the pull request:
1.finish setting page except changing user's picture
2.user data also save in a json file and it can be loaded into settingpage
3.change layout's title

KJay221 committed:
change settingpage

coliver0221 commented on src/layouts/generalLayout.tsx:
```diff
@@ -4,14 +4,20 @@ import Navbar from '../components/nav';
+const GeneralLayout: React.FC<IProps> = ({ children, title }) => {
   const className = classnames('general-layout');
+  const titleClass = classnames('general-title');
+  const rightContainer = classnames('general-container');
```
Why you name it `rightContainer` ?

KJay221 replied:
I change it to `container`

coliver0221 commented on src/layouts/generalLayout.tsx:
```diff
@@ -4,14 +4,20 @@ import Navbar from '../components/nav';
+  const titleClass = classnames('general-title');
+  const rightContainer = classnames('general-container');
   return (
     <main className={className}>
```
Since you just assign a constant value to it. I think you can just assign className here without define variables.
It may be like this:
```tsx
<main className={classnames('general-layout')}>
```
Same recommendations for className assign below.

KJay221 replied:
done

coliver0221 commented on src/scss/layout/_generalLayout.scss:
```diff
@@ -1,10 +1,66 @@
 
-@media all and (max-width: 720px) {
+.general-title{
+  margin-top: 90px;
```
It will be better if you separate general setting and customized setting for specific width range.
It may be like this:
```scss
.general-title {
  width: auto;
  height: auto;
  text-align: left;
  color: rgb(112, 112, 112);
}

// 1280 < size
@media all and (min-width: 1280px) {
  .general-title {
    margin-top: 90px;
    margin-left: 60px;
    font-size: 40px;
  }
}
```

KJay221 replied:
done

coliver0221 commented:
Please pull the latest version from origin master and resolve the conflicts.

KJay221 committed:
style: fix PR problem

fix change settingpage PR problem

coliver0221 commented on src/components/settingpage/rightcomponent/index.tsx:
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import classnames from 'classnames';
+import data from './test.json';
```
You should call API to get all these data.
There is a example you can follow in `src/components/nav/user.tsx`.
Because the API will also give you the image url you need in `LeftComponent`, I think you can call the API in `src/components/settingpage/index.tsx` and pass information to child components as parameters.
For more information you can check [API Doc](https://et01.docs.apiary.io/#reference).

KJay221 committed:
feat: add change password

add change password dialog

KJay221 committed:
feat: add settingpage i18n

add settingpage i18n but when turning to english version will have some typesetting problems itwill
be fixed in RWD commit

KJay221 committed:
style: fix lint problems

fix lint problem in setting page and loginform

KJay221 committed:
feat: 1920~450 width RWD

finish settingpage 1920~450 width RWD and fix some css problems

KJay221 committed:
feat: finish most of API and fix css problems

connect most of API but some API have problems(img,account,password,identification code). And I use
js to do text overflow problem rather than scss because scss may lead to some RWD problems.By the
way,amis just have 1 data can be tested.

KJay221 committed:
feat: add layout i18n and fix some lint problem

add layout title i18n and fix some lint problems

coliver0221 commented on src/components/settingpage/leftcomponent/img.tsx:
```diff
@@ -0,0 +1,22 @@
+      <img
+        className="setting-left--img"
+        alt="can't show"
+        src="https://i.imgur.com/1T8X1XS.png"
```
You should replace this fixed img source with the url return by API.

KJay221 replied:
done

KJay221 committed:
feat: change password dialog RWD

add change password dialog RWD

KJay221 committed:
feat: add 450 width and fix some css problems

add 450 width and fix some css problems

coliver0221 commented on src/components/settingpage/index.tsx:
```diff
@@ -0,0 +1,121 @@
+    if (response.status === 200) {
+      // fetch success
+      const data = await response.json();
+      data.map((item: string) =>
```
Why you define item as a string?
It should be a ts object.

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/index.tsx:
```diff
@@ -0,0 +1,121 @@
+
+  useEffect(() => {
+    (async () => {
+      await fetchUser();
```
You can do `await fetchAmis()` here without define a new `useEffect()`.

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/content.tsx:
```diff
@@ -0,0 +1,84 @@
+  const datacontainer = classnames('setting-left--datacontainer');
+  const content = classnames('setting-left--datacontent');
+  const title = classnames('setting-left--datatitle');
+  const buttonStyle = classnames('setting-left--buttonStyle');
```
Class name problems.
1. `-` but not `--`
2. If class name is constant you do not need to define it as variables

KJay221 replied:
done

KJay221 replied:
Some class name use more than one time. So I set it to string

coliver0221 commented on src/components/settingpage/leftcomponent/content.tsx:
```diff
@@ -0,0 +1,84 @@
+  const [dialogState, setDialog] = useState<boolean>(false);
+
+  return (
+    <div className={classnames('setting-left--contentcontainer')}>
```
Class name problems.
1. `-` but not `--`
2. If class name is constant you do not need to define it as variables

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/dialogbox.tsx:
```diff
@@ -0,0 +1,110 @@
+  const { t } = useTranslation();
+
+  const title = classnames('setting-left--dialogtitle');
+  const passwordinput = classnames('setting-left--passwordInput');
```
Class name problems.
1. `-` but not `--`
2. If class name is constant you do not need to define it as variables

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/dialogbox.tsx:
```diff
@@ -0,0 +1,110 @@
+    changeconfirmPassword('');
+  }
+
+  return (
```
4 class name problems in this part.
1. `-` but not `--`
2. If class name is constant you do not need to define it as variables

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/img.tsx:
```diff
@@ -0,0 +1,22 @@
+  username: string;
+}
+
+const Img: React.FC<IUserImg> = ({ img, username }) => {
```
3 class name problems in this part.
1. `-` but not `--`
2. If class name is constant you do not need to define it as variables

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/index.tsx:
```diff
@@ -0,0 +1,27 @@
+  ethAddress,
+}) => {
+  return (
+    <div className={classnames('setting-left--container')}>
```
Class name problems.
1. `-` but not `--`
2. If class name is constant you do not need to define it as variables

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/rightcomponent/index.tsx:
```diff
@@ -0,0 +1,98 @@
+  const listItems = IAmisList.map((content) => {
+    const code = `${content.id.substr(0, 32)}...`;
+
+    return (
```
Many class name problems in this part.
1. `-` but not `--`
2. If class name is constant you do not need to define it as variables

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/rightcomponent/index.tsx:
```diff
@@ -0,0 +1,98 @@
+    );
+  });
+
+  return (
```
Many class name problems in this part.
1. `-` but not `--`
2. If class name is constant you do not need to define it as variables

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/content.tsx:
```diff
@@ -0,0 +1,84 @@
+        <div className={title}>{t('settingpage.account')}</div>
+        <div className={content}>{data[0].account}</div>
+      </div>
+      <div id="passwordcontainer1">
```
Why you use `id` but not `class name` here?

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/content.tsx:
```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import { useTranslation } from 'react-i18next';
+import data from './test.json';
```
You should replace this testing data with data return by API after API fixed.

KJay221 replied:
API did not give the text number of password so I need to use local data to show

coliver0221 replied:
Ok. Then you should replace this after the API has been modified.

coliver0221 commented on src/components/settingpage/leftcomponent/dialogbox.tsx:
```diff
@@ -0,0 +1,110 @@
+  const { t } = useTranslation();
+
+  const title = classnames('setting-left--dialogtitle');
+  const passwordinput = classnames('setting-left--passwordInput');
```
Please use lower camel case.

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/dialogbox.tsx:
```diff
@@ -0,0 +1,110 @@
+          Authorization: `Bearer ${user.bearer}`,
+          'Content-Type': 'application/json',
+        }),
+        body: password,
```
You can put JSON.stringify(...) here without define it as variable.
It may look like this :
```tsx
body: JSON.stringify(
  {
    ...
  }
),
```

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/dialogbox.tsx:
```diff
@@ -0,0 +1,110 @@
+  changestate: (display: boolean) => void;
+}
+
+const DialogBox: React.FC<IProps> = ({ changestate }) => {
```
Please use lower camel case.

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/dialogbox.tsx:
```diff
@@ -0,0 +1,110 @@
+
+  const [oldPassword, changeoldPassword] = useState<string>('');
+  const [newPassword, changenewPassword] = useState<string>('');
+  const [confirmPassword, changeconfirmPassword] = useState<string>('');
```
Please use lower camel case.

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/dialogbox.tsx:
```diff
@@ -0,0 +1,110 @@
+    );
+    if (response.status === 200) {
+      // fetch success
+      window.alert(t('settingpage.successChange'));
```
If you need to use alert, you need to disable eslint rule for this line.
You can do this by hovering the warning and click `Quick fix` then choose `Disable ... for this line`.
Same comments for alert below.

KJay221 replied:
done

coliver0221 commented on src/i18n/en-US.json:
```diff
@@ -15,5 +15,38 @@
+    "newPassword": "Enter new password",
+    "confirmPassword": "Enter new password again",
+    "changePassword": "Change Password",
+    "changelanguage": "中文",
```
Don't set Chinese words in English setting file.

KJay221 replied:
done

coliver0221 commented on src/i18n/zh-TW.json:
```diff
@@ -15,5 +15,38 @@
+    "newPassword": "輸入新密碼",
+    "confirmPassword": "再次輸入新密碼",
+    "changePassword": "更改密碼",
+    "changelanguage": "ENGLISH",
```
Don't set English words in Chinese setting file.

KJay221 replied:
done

KJay221 committed:
fix: fix problems

fix problems mentioned in github

coliver0221 commented on src/components/settingpage/index.tsx:
```diff
@@ -0,0 +1,119 @@
+    if (response.status === 200) {
+      // fetch success
+      const data = await response.json();
+      data.map((item: object) =>
```
Since you don't actually map the array, it's better to use `forEach` instead of `map` here.

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/index.tsx:
```diff
@@ -0,0 +1,119 @@
+      // fetch success
+      const data = await response.json();
+      data.map((item: object) =>
+        addAMis(Object.values(item)[2], Object.values(item)[1]),
```
Since the `item` is an object, you can use key to access the data for more readability.
It may be like this:
```tsx
addAMis(item.name, item.value)
```

KJay221 replied:
done

coliver0221 commented on src/components/settingpage/leftcomponent/dialogbox.tsx:
```diff
@@ -0,0 +1,112 @@
+
+  return (
+    <div className={classnames('setting-left-background')}>
+      <div className={classnames('setting-left-dialogcontainer')}>
```
The content will exceed the container vertically.
Please fix it.

KJay221 replied:
done

KJay221 committed:
fix: fix problems

fix problems mentioned on github

coliver0221 commented:
LGTM.
